main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -75,7 +75,30 @@ object Safety {
     val renv = def0.spec.tparams.map(_.sym).foldLeft(RigidityEnv.empty) {
       case (acc, e) => acc.markRigid(e)
     }
-    visitExp(def0.impl.exp, renv, root)
+    visitTestEntryPoint(def0, root) ::: visitExp(def0.impl.exp, renv, root)
+  }
+
+  /**
+    * Checks that if `def0` is a test entry point that it is well-behaved.
+    */
+  private def visitTestEntryPoint(def0: Def, root: Root): List[CompilationMessage] = {
```

Comment:
Lets immediately do this test: `def0.spec.ann.isTest`

if (def0.spec.ann.isTest) {... } else {Nil}